test/jdk/tools/jpackage/helpers/jdk/jpackage/test/AdditionalLauncher.java

```diff
@@ -232,6 +233,44 @@ private void verify(JPackageCommand cmd) throws IOException {
         .executeAndVerifyOutput();
     }
 
+    public void verifyPackageInstalled(JPackageCommand cmd) {
+        final String formatString;
+        if (cmd.isPackageUnpacked()) {
+            formatString = "Verify unpacked: %s";
+        } else {
+            formatString = "Verify installed: %s";
+        }
+        TKit.trace(String.format(formatString, cmd.getPrintableCommandLine()));
+
+        if (!cmd.isRuntime()) {
+            if (PackageType.WINDOWS.contains(cmd.packageType())
+                    && !cmd.isPackageUnpacked(
+                            "Not verifying desktop integration")) {
+                new WindowsHelper.DesktopIntegrationVerifier(cmd, name);
+            }
+        }
+    }
+
+    public void verifyPackageUninstalled(JPackageCommand cmd) {
+        TKit.trace(String.format("Verify uninstalled: %s",
+                cmd.getPrintableCommandLine()));
+        if (!cmd.isRuntime()) {
+            TKit.assertPathExists(cmd.appLauncherPath(), false);
+
+            if (PackageType.WINDOWS.contains(cmd.packageType())) {
+                new WindowsHelper.DesktopIntegrationVerifier(cmd, name);
+            }
+        }
+
+        Path appInstallDir = cmd.appInstallationDirectory();
+        if (TKit.isLinux() && Path.of("/").equals(appInstallDir)) {
+            ApplicationLayout appLayout = cmd.appLayout();
+            TKit.assertPathExists(appLayout.runtimeDirectory(), false);
+        } else {
+            TKit.assertPathExists(appInstallDir, false);
+        }
```

Comment:
I don't understand what is going on here and why branching is required. Also seems like verifyPackageUninstalled() function is never called.